static/js/calculateSceneLength.js

```diff
@@ -0,0 +1,87 @@
+var _ = require('ep_etherpad-lite/static/js/underscore');
+
+var shared = require('./shared');
+var utils = require('./utils');
+
+var calculateSceneLength = function(attributeManager, rep, editorInfo) {
+  this.attributeManager = attributeManager;
+  this.rep = rep;
+  this.editorInfo = editorInfo;
+  this.thisPlugin = pad.plugins.ep_script_elements;
+  this.calculateSceneEdgesLength = thisPlugin.calculateSceneEdgesLength;
+};
+
+calculateSceneLength.prototype.run = function() {
+  var lineDefaultSize = utils.getLineDefaultSize();
+  var $headings = utils.getPadInner().find('div:has(heading)');
+
+  // when script is disable on EASC we don't calculate the scene length
+  var headingsAreVisible = this.thisPlugin.isScriptActivated;
+  if (!headingsAreVisible) {
+    return;
+  }
+  var $sceneIntervals = this._getSceneIntervals($headings);
+  var scenesLength = _.map($sceneIntervals, function($sceneInterval) {
+    var sceneLength = this._getSceneLength($sceneInterval, lineDefaultSize);
+    return sceneLength;
+  }, this);
+  this._applyAttributeOnScene($headings, scenesLength);
+};
+
+calculateSceneLength.prototype._applyAttributeOnScene = function($headings, scenesLength) {
+  _.each($headings, function(heading, index){
+    this._applyAttributeOnLineIfNecessary(heading, scenesLength[index]);
+  }, this);
+};
+
+calculateSceneLength.prototype._applyAttributeOnLineIfNecessary= function(element, attribValue) {
+  var self = this;
+  self.editorInfo.ace_inCallStackIfNecessary('nonundoable', function(){
+    // we check the script element attribute value to avoid applying a line
+    // attribute on a line that is not heading. E.g. on tests is common to have
+    // something like '<div><action>...<br><heading>...<br><other>...<br>',
+    // what is not a 'heading' but it has the '<heading>'
+    var $element = $(element);
+    var line =  utils.getLineNumberFromDOMLine($element, self.rep);
+    var scriptElementType = utils.getLineType(line, self.attributeManager);
+    var isHeading = scriptElementType === 'heading';
+
+    if (self._sceneLengthChanged(line, attribValue) && isHeading) {
+      self.attributeManager.removeAttributeOnLine(line, shared.SCENE_LENGTH_ATTRIB_NAME);
+      self.attributeManager.setAttributeOnLine(line, shared.SCENE_LENGTH_ATTRIB_NAME, attribValue);
+    }
+  });
+};
+
+calculateSceneLength.prototype._sceneLengthChanged = function(lineNumber, sceneLengthUpdated) {
+  var actualSceneLengthValue = this.attributeManager.getAttributeOnLine(lineNumber, shared.SCENE_LENGTH_ATTRIB_NAME);
+  return Number(actualSceneLengthValue ) !== sceneLengthUpdated;
+};
+
+calculateSceneLength.prototype._getSceneIntervals = function($headings) {
+  return _($headings).map(function(line){
```

Comment:
não sei se vc sabe, mas [o jQuery tb tem `map`](https://api.jquery.com/map/). Dá para evitar o `require` do `underscore` quando vc só for usar `_.map` no arquivo...